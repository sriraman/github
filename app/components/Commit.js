import React, { Component } from 'react'
import { View, Text, Image } from 'react-native'
import commitStyles from '../styles/commit'
import moment from 'moment'

class Commit extends Component {
	render () {
		return <View style={commitStyles.container}>
			{this.props.data.author &&
				<View style={commitStyles.authorContainer}>
					<Image source={{ uri: this.props.data.author.avatar_url }} style={commitStyles.avatar} />
					<Text style={commitStyles.authorName}>{this.props.data.author.login}</Text>
				</View>
			}
			<Text> {this.props.data.commit.message.split('\n')[0]} </Text>
			<Text style={commitStyles.timeAgo}> {moment(this.props.data.commit.author.date).fromNow()} </Text>
		</View>
	}
}

module.exports = Commit
