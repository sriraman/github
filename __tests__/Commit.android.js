import 'react-native'
import React from 'react'
import Commit from '../app/components/Commit'
import moment from 'moment'

import renderer from 'react-test-renderer'

/**
 * Commit Component with All Data
 */
it('render Commit component correctly', () => {
	let commitData = {
		author: {
			avatar_url: 'http://placehold.it/25/25',
			login: 'sriraman'
		},
		commit: {
			message: 'Abc',
			author: {
				date: moment()
			}
		}
	}
	let commit = renderer.create(
		<Commit data={commitData}/>
	).toJSON()
	expect(commit).toMatchSnapshot()
})

/**
 * Commit Component without Author data (Author data is optional in Github API)
 */
it('render Commit component correctly even without author data', () => {
	let commitData = {
		commit: {
			message: 'Abc',
			author: {
				date: moment()
			}
		}
	}
	let commit = renderer.create(
		<Commit data={commitData}/>
	).toJSON()
	expect(commit).toMatchSnapshot()
})
