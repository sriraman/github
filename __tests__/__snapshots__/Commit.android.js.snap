// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Commit component correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "justifyContent": "center",
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://placehold.it/25/25",
        }
      }
      style={
        Object {
          "borderRadius": 13,
          "height": 25,
          "width": 25,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "fontWeight": "bold",
          "paddingLeft": 10,
        }
      }
    >
      sriraman
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     
    Abc
     
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#ccc",
      }
    }
  >
     
    a few seconds ago
     
  </Text>
</View>
`;

exports[`render Commit component correctly even without author data 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "justifyContent": "center",
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     
    Abc
     
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#ccc",
      }
    }
  >
     
    a few seconds ago
     
  </Text>
</View>
`;
